lib/SIL/Utils/PrunedLiveness.cpp

```diff
@@ -442,32 +442,66 @@ template class PrunedLiveRange<MultiDefPrunedLiveness>;
 //                             SSAPrunedLiveness
 //===----------------------------------------------------------------------===//
 
-void SSAPrunedLiveness::findBoundariesInBlock(
-    SILBasicBlock *block, bool isLiveOut,
-    PrunedLivenessBoundary &boundary) const {
-  assert(isInitialized());
-
-  // For SSA, a live-out block cannot have a boundary.
-  if (isLiveOut)
-    return;
-
-  bool isDefBlockState = isDefBlock(block);
+/// Given live-within (non-live-out) \p block, find the last user.
```

Comment:
Would it be worth asserting that `liveness.getBlockLiveness(block) == ::LiveWithin` at the top?

Reply:
I can't argue with that now that I already said "it's easy for someone to use it wrong"